web/src/stream_settings_ui.js

```diff
@@ -792,7 +794,7 @@ export function switch_rows(event) {
     const row_data = get_row_data($switch_row);
     if (row_data) {
         const stream_id = row_data.id;
-        switch_to_stream_row(stream_id);
+        switch_to_stream_row(stream_id, "general");
     } else if (event === "up_arrow" && !row_data) {
         $("#search_stream_name").trigger("focus");
     }
```

Comment:
If I visit http://localhost:9991/#streams/12/support/subscribers, and repeatedly click on that stream's row in the left side, it very clearly has the URL change twice each time.

Reply:
Ah okay so there's a few things going on.

---

One is that the url is updating from two places when clicking a stream on the left side:

<img width="576" alt="image" src="https://github.com/zulip/zulip/assets/5634097/19667bde-6a1a-4d93-adfa-0a69d1654716">

I tried removing the first call, but then the second call never happens because it gets cut off here because the url hasn't changed yet.

```js
export function update_toggler_for_sub(sub) {
    if (!hash_parser.is_editing_stream(sub.stream_id)) {
        return;
    }
```

And we definitely need the second call because that's what updates the url when a user clicks on the tabs themselves to change it.

So I think it's fine to just keep both of them.

---

But we do want them to call the same thing! `open_edit_panel_for_row(this, "general");` was causing the general tab to open, but then `update_toggler_for_sub` uses a pre-saved `stream_edit_toggler.select_tab` instead. We could pick either, but I decided to go with existing logic from before this PR and always open the `stream_edit_toggler.select_tab`. So here's what I ended up changing to fix this in the end:

```diff
diff --git a/web/src/stream_edit.js b/web/src/stream_edit.js
index 1f20e345c5..ad3e62df5b 100644
--- a/web/src/stream_edit.js
+++ b/web/src/stream_edit.js
@@ -548,7 +548,7 @@ export function initialize() {
         stream_settings_components.sub_or_unsub(sub, $stream_row);
 
         if (!sub.subscribed) {
-            open_edit_panel_for_row($stream_row, "general");
+            open_edit_panel_for_row($stream_row, stream_edit_toggler.select_tab);
         }
         stream_ui_updates.update_regular_sub_settings(sub);
 
@@ -616,7 +616,7 @@ export function initialize() {
 
     $("#streams_overlay_container").on("click", ".stream-row", function (e) {
         if ($(e.target).closest(".check, .subscription_settings").length === 0) {
-            open_edit_panel_for_row(this, "general");
+            open_edit_panel_for_row(this, stream_edit_toggler.select_tab);
         }
     });
 ```

Reply:
Pushing with this change now.

Reply:
That looks reasonable to me!